discovery-provider/src/tasks/entity_manager/social_features.py

```diff
@@ -56,144 +56,147 @@ def create_social_record(params: ManageEntityParameters):
 
     validate_social_feature(params)
 
+    record_types = action_to_record_types[params.action]
     create_record: Union[Save, Follow, Repost, Subscription, None] = None
-    if params.action == Action.FOLLOW:
-        create_record = Follow(
-            blockhash=params.event_blockhash,
-            blocknumber=params.block_number,
-            created_at=params.block_datetime,
-            txhash=params.txhash,
-            follower_user_id=params.user_id,
-            followee_user_id=params.entity_id,
-            is_current=True,
-            is_delete=False,
-        )
-    if params.action == Action.SAVE:
-        create_record = Save(
-            blockhash=params.event_blockhash,
-            blocknumber=params.block_number,
-            created_at=params.block_datetime,
-            txhash=params.txhash,
-            user_id=params.user_id,
-            save_item_id=params.entity_id,
-            save_type=params.entity_type.lower(),
-            is_current=True,
-            is_delete=False,
-        )
-    if params.action == Action.REPOST:
-        create_record = Repost(
-            blockhash=params.event_blockhash,
-            blocknumber=params.block_number,
-            created_at=params.block_datetime,
-            txhash=params.txhash,
-            user_id=params.user_id,
-            repost_item_id=params.entity_id,
-            repost_type=params.entity_type.lower(),
-            is_current=True,
-            is_delete=False,
-        )
-    if params.action == Action.SUBSCRIBE:
-        create_record = Subscription(
-            blockhash=params.event_blockhash,
-            blocknumber=params.block_number,
-            created_at=params.block_datetime,
-            txhash=params.txhash,
-            user_id=params.entity_id,
-            subscriber_id=params.user_id,
-            is_current=True,
-            is_delete=False,
-        )
-
-    if create_record:
-        params.add_social_feature_record(
-            params.user_id,
-            params.entity_type,
-            params.entity_id,
-            action_to_record_type[params.action],
-            create_record,
-        )
+    for record_type in record_types:
+        if record_type == EntityType.FOLLOW:
+            create_record = Follow(
```

Comment:
since this follow / subscription coupling on create is an exception, is it easier to just add a Subscription record here?

Reply:
i actually found it easier to edit `action_to_record_type` so we can collect existing subscriptions in `entity_manger.py` to invalidate without any special conditional logic on follow or unfollow